src/Emitter.ts

```diff
@@ -486,6 +487,46 @@ export default class Emitter {
             t = t.replace(/\)$/, '');
         }
 
+        function _resolvePromiseObject(ty: string): string {
+            const matches = ty.match(rgxObjectType);
+            if (matches && matches[1] && matches[2]) {
+                const indexTypeStr = matches[1].trim();
+                const valueTypeStr = matches[2].trim();
+
+                if (indexTypeStr !== 'string' && indexTypeStr !== 'number') {
+                    warn(`Invalid object index type: "${matches[1]}", must be "string" or "number". Falling back to "Object<any, any>".`);
+                    return `Object<any, any>`;
+                }
+
+                return `{ [key: ${indexTypeStr}]: ${valueTypeStr} }`;
+            }
+        }
+
+        // try promise type
+        if (t.startsWith('Promise.<')) {
+            const matches = t.match(rgxPromiseType);
+            if (matches) {
+                if (matches[1].startsWith('Object.<')) {
+                    return dom.create.namedTypeReference(_resolvePromiseObject(matches[1]));
+                } else if (matches[1].startsWith('Array')) {
+                    const matchesArray = matches[1].match(rgxArrayType);
+                    if (matchesArray && matchesArray[1]) {
+                        if (rgxObjectType.test(matchesArray[1])) {
+                            return dom.create.namedTypeReference(`Promise<Array<${_resolvePromiseObject(matchesArray[1])}>>`);
+                        } else {
+                            return dom.create.namedTypeReference(`Promise<${matches[1].replace(/(\.)/g, '')}>`);
+                        }
+                    } else {
+                        return dom.create.namedTypeReference(`Promise<${matches[1].replace(/(\.)/g, '')}>`);
+                    }
+                } else {
+                    return dom.create.namedTypeReference(`Promise<${matches[1].replace(/(\.)/g, '')}>`);
+                }
+            } else {
+                return dom.create.namedTypeReference(`Promise<any>`);
+            }
+        }
```

Comment:
Ok. Would you be ok with merging my changes till we can get a proper fix done up? because even though this implementation is rather quick and dirty and I am naive to the internal workings of this project, some users of this library would appreciate that there is some support rather than none at all.

Reply:
I will still love to help out, just for now, I dont know how long it would take to add proper implementation for this in downstream modules on top of the fact, I dont know Typescript well enough that I could even help with adding support to dts-dom, and partial support is better than nothing.

Reply:
Yeah I have been doing research a bit and as of October this year the Typescript compiler finally has a good AST/Emit API. So I'm going to rewrite the lib to use the typescript compiler and not strings via dts-dom, which will solve this problem.

I'm not going to merge this as-is because there are some changes not related to doing Promise<T> in here, but you can easily use a git url in your package.json to point to this sha if you need this feature now.

I don't think it will take too long for me to do the rewrite, this lib is pretty small.